packages/relay/tests/lib/eth.spec.ts

```diff
@@ -962,13 +989,31 @@ describe('Eth calls using MirrorNode', async function () {
       expect(resCached).to.equal(EthImpl.zeroHex);
     });
 
-    it('should return the bytecode', async () => {
-      const bytecode = '0x608060405234801561001057600080fd5b5060405161078938038061078983398181016040528101906100329190';
+    it('should return the deployed_bytecode from the mirror node', async () => {
+      mock.onGet(`contracts/${contractAddress1}`).reply(200, defaultContract);
+      sdkClientStub.getContractByteCode.returns(Buffer.from(deployedBytecode.replace('0x', ''), 'hex'));
+
+      const res = await ethImpl.getCode(contractAddress1, null);
+      expect(res).to.equal(mirrorNodeDeployedBytecode);
+    });
+
+    it('should return the bytecode from SDK if Mirror Node returns 404', async () => {
+      mock.onGet(`contracts/${contractAddress1}`).reply(404, defaultContract);
+      sdkClientStub.getContractByteCode.returns(Buffer.from(deployedBytecode.replace('0x', ''), 'hex'));
+
+      const res = await ethImpl.getCode(contractAddress1, null);
+      expect(res).to.equal(deployedBytecode);
+    });
 
-      sdkClientStub.getContractByteCode.returns(Buffer.from(bytecode.replace('0x', ''), 'hex'));
+    it('should return the bytecode from SDK if Mirror Node returns empty deployed_bytecode', async () => {
```

Comment:
Missing one test for when mirror node returns with `contract.runtime_bytecode === EthImpl.emptyHex`